p256-cm4/src/lib.rs

```diff
@@ -1,8 +1,17 @@
 #![no_std]
 #![allow(clippy::missing_safety_doc)]
 
+mod new_api;
 mod sys;
 
+#[cfg(feature = "p256")]
+mod p256;
+
+#[cfg(feature = "signature")]
```

Comment:
This also needs `p256` to compile.  Not sure if there's a way to break that dependency.

Tested with `cargo build --features signature --target thumbv7em-none-eabi`